boards/sma_q3/src/main.rs

```diff
@@ -0,0 +1,472 @@
+    )
+    .finalize(components::ieee802154_component_helper!(
+        nrf52840::ieee802154_radio::Radio,
+        nrf52840::aes::AesECB<'static>
+    ));
+
+    // Not exposed in favor of the BMP280, but present.
+    // Possibly needs power management all the same.
+    let _temp = components::temperature::TemperatureComponent::new(
+        board_kernel,
+        capsules::temperature::DRIVER_NUM,
+        &base_peripherals.temp,
+    )
+    .finalize(());
+
+    let sensors_i2c_bus = static_init!(
+        capsules::virtual_i2c::MuxI2C<'static>,
+        capsules::virtual_i2c::MuxI2C::new(&base_peripherals.twi1, None, dynamic_deferred_caller)
+    );
+
+    base_peripherals.twi1.configure(
+        nrf52840::pinmux::Pinmux::new(I2C_TEMP_SCL_PIN as u32),
+        nrf52840::pinmux::Pinmux::new(I2C_TEMP_SDA_PIN as u32),
+    );
+    base_peripherals.twi1.set_master_client(sensors_i2c_bus);
+
+    let bmp280 = Bmp280Component::new(sensors_i2c_bus, mux_alarm)
+        .finalize(bmp280_component_helper!(nrf52840::rtc::Rtc<'static>));
+
+    let temperature = components::temperature::TemperatureComponent::new(
+        board_kernel,
+        capsules::temperature::DRIVER_NUM,
+        bmp280,
+    )
+    .finalize(());
+
+    let rng = components::rng::RngComponent::new(
+        board_kernel,
+        capsules::rng::DRIVER_NUM,
+        &base_peripherals.trng,
+    )
+    .finalize(());
+
+    // Initialize AC using AIN5 (P0.29) as VIN+ and VIN- as AIN0 (P0.02)
+    // These are hardcoded pin assignments specified in the driver
+    let analog_comparator = components::analog_comparator::AcComponent::new(
+        &base_peripherals.acomp,
+        components::acomp_component_helper!(
+            nrf52840::acomp::Channel,
+            &nrf52840::acomp::CHANNEL_AC0
+        ),
+        board_kernel,
+        capsules::analog_comparator::DRIVER_NUM,
+    )
+    .finalize(components::acomp_component_buf!(
+        nrf52840::acomp::Comparator
+    ));
+
+    nrf52_components::NrfClockComponent::new(&base_peripherals.clock).finalize(());
+
+    let scheduler = components::sched::round_robin::RoundRobinComponent::new(&PROCESSES)
+        .finalize(components::rr_component_helper!(NUM_PROCS));
+
+    let periodic_virtual_alarm = static_init!(
+        capsules::virtual_alarm::VirtualMuxAlarm<'static, nrf52840::rtc::Rtc>,
+        capsules::virtual_alarm::VirtualMuxAlarm::new(mux_alarm)
+    );
+    periodic_virtual_alarm.setup();
+
+    let platform = Platform {
+        temperature,
+        button,
+        ble_radio,
+        ieee802154_radio,
+        pconsole,
+        console,
+        led,
+        gpio,
+        rng,
+        alarm,
+        analog_comparator,
+        ipc: kernel::ipc::IPC::new(
+            board_kernel,
+            kernel::ipc::DRIVER_NUM,
+            &memory_allocation_capability,
+        ),
+        scheduler,
+        systick: cortexm4::systick::SysTick::new_with_calibration(64000000),
+    };
+
+    fn load_processes(
```

Comment:
Can you comment on how this helps debugging? Other board maintainers might be interested.

Reply:
Oh, and I made use of this for the "fake off" functionality, where if a button is pressed, processes are stopped, but when pressed again, they are loaded anew. This may not be needed any more after https://github.com/tock/tock/pull/3068 lands.

Reply:
Can you add this as a comment in the main.rs file?

Reply:
Done, next to the procedure itself.